### ISSUE TITLE
Updates to Notification component

### ISSUE THREAD

**Issue by Elanhant** [gatsby-inc/gatsby-interface#239]
Most of this is based on [these comments](https://github.com/gatsby-inc/gatsby-interface/pull/237#pullrequestreview-373975189):
- [x] Use `theme.space[2]` for `borderLeft` width in "PRIMARY" variant
- [x] Use `theme.space[6]` for `paddingLeft` width in "PRIMARY" variant for "desktop" media query
- [x] Nudge variant colors a bit (making the BRAND ones a bit friendlier and calmer)
- [x] Adjust icon sizes (the size that the icons are rendered in is 👍, but all other icons we are currently using contain inner padding, while our custom ones do not)
- [x] _(Optional)_ use [Storybook Figma Addon](https://www.figma.com/resources/extensions-and-apis/storybook-figma-addon/) to include Notification designs in Storybook.

**Comment by fk** [comment]
Thanks @Elanhant for capturing my ramblings! :pray: 
I'll grab this! :v: … let's see if I can get my head around the optional item on the list! :)

**Comment by mfrachet** [comment]
We have another issue with the Navigation that could potentially make the things a bit harder:

In the navigation, we have a left side and a right side:

<img width="1920" alt="Screenshot 2020-03-24 at 10 33 21" src="https://user-images.githubusercontent.com/3874873/77410259-eade4000-6dba-11ea-96a8-556284e59515.png">

And we do have a semantic and accessible issue: there's a `div` in the middle of a `<ul>` element. This is because of the `Navigation.Spacer` component.

I have an idea to fix this: having two lists, one for the left side and one for the right side. This would force us to break the entire Navigation API (more precisely the "BaseNavigation") because it owns the main "nav" HTML tag.

However, we could potentially have a version + 1 of Navigation that could like:

```jsx
 <Navigation>
      <NavigationList>
        <NavigationItem>Home</NavigationItem>
        <NavigationItem>Pricing</NavigationItem>
      </NavigationList>

      <NavigationList align="right">
        <NavigationItem>Get started</NavigationItem>
        <NavigationItem>Other action</NavigationItem>
      </NavigationList>
    </Navigation>
```

This would keep the HTML semantic and allows to fix the a11y issue. 


What do you think of this? @Elanhant @amberleyromo @joshwcomeau @greglobinski @fk ?

**Comment by Elanhant** [comment]
@mfrachet you probably wanted to post this to #246 🙂 

I think we can use `<li role="separator" />` instead of this div (looks like it is allowed, see [Permitted ARIA roles](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/li)).

**Comment by fk** [comment]
Fixed in df3f602d479a5a6482d297f660a89885e182bb89. Also ref. #242, #237.
